### PR TITLE
Correct traceback caused by None "target_id"

### LINKED ISSUE #276: Traceback occurs when ran with factory default firmware on Realtek RTL8195AM
As posted by @MarceloSalazar here: https://github.com/ARMmbed/mbed-os-example-blinky/issues/100#issuecomment-350307215

The issue apparently goes away when the firmware is updated to the latest DAPLink.

(logs are attached since they don't display that great here)
[log.txt](https://github.com/ARMmbed/mbed-ls/files/1543249/log.txt)


```

$ mbedls --debug
DEBUG:mbedls.main:mbed-ls ver. 1.3.4
DEBUG:mbedls.main:host: ('posix', 'Darwin', '15.6.0', 'Darwin Kernel Version 15.6.0: Mon Aug 29 20:21:34 PDT 2016; root:xnu-3248.60.11~1/RELEASE_X86_64', 'darwin')
DEBUG:mbedls.lstools_darwin:finding in 
{ 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'AppleUSBHostLegacyClient'},
                               { 'IORegistryEntryName': 'AppleUSBCDCCompositeDevice'},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'BSD Name': 'disk2',
                                                                                                                                                                                                                                                                                         'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOMediaBSDClient'}],
                                                                                                                                                                                                                                                                                         'IORegistryEntryName': 'MBED microcontroller Media'}],
                                                                                                                                                                                                                                                          'IORegistryEntryName': 'IOBlockStorageDriver'}],
                                                                                                                                                                                                                           'IORegistryEntryName': 'IOBlockStorageServices'}],
                                                                                                                                                                                            'IORegistryEntryName': 'IOSCSIPeripheralDeviceType00'}],
                                                                                                                                                             'IORegistryEntryName': 'IOSCSILogicalUnitNub'}],
                                                                                                                              'IORegistryEntryName': 'IOUSBMassStorageDriver'}],
                                                                                               'IORegistryEntryName': 'IOUSBMassStorageDriverNub',
                                                                                               'idProduct': 516,
                                                                                               'idVendor': 3368}],
                                                                'IORegistryEntryName': 'IOUSBMassStorageInterfaceNub',
                                                                'idProduct': 516,
                                                                'idVendor': 3368}],
                                 'IORegistryEntryName': 'USB_MSC',
                                 'idProduct': 516,
                                 'idVendor': 3368},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'AppleUSBACMControl'}],
                                 'IORegistryEntryName': '1050v9.2098969D09E948678',
                                 'idProduct': 516,
                                 'idVendor': 3368},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IODialinDevice': '/dev/tty.usbmodem1412',
                                                                                                                              'IORegistryEntryName': 'IOSerialBSDClient'}],
                                                                                               'IORegistryEntryName': 'IOModemSerialStreamSync'}],
                                                                'IORegistryEntryName': 'AppleUSBACMData',
                                                                'idProduct': 516,
                                                                'idVendor': 3368}],
                                 'IORegistryEntryName': 'USB_CDC1',
                                 'idProduct': 516,
                                 'idVendor': 3368},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDInterface'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'}],
                                                                'IORegistryEntryName': 'IOUSBHostHIDDevice'}],
                                 'IORegistryEntryName': 'MBED CMSIS-DAP',
                                 'idProduct': 516,
                                 'idVendor': 3368}],
  'IORegistryEntryName': 'MBED CMSIS-DAP',
  'idProduct': 516,
  'idVendor': 3368}
DEBUG:mbedls.lstools_darwin:found usb info {'tty': None, 'serial': None, 'vendor_id': 3368, 'product_id': 516}
DEBUG:mbedls.lstools_darwin:finding in 
{ 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'AppleUSBHostLegacyClient'},
                               { 'IORegistryEntryName': 'BroadcomBluetoothHostControllerUSBTransport',
                                 'idProduct': 33424,
                                 'idVendor': 1452},
                               { 'IORegistryEntryName': 'IOUSBHostInterface',
                                 'idProduct': 33424,
                                 'idVendor': 1452},
                               { 'IORegistryEntryName': 'IOUSBHostInterface',
                                 'idProduct': 33424,
                                 'idVendor': 1452},
                               { 'IORegistryEntryName': 'IOUSBHostInterface',
                                 'idProduct': 33424,
                                 'idVendor': 1452},
                               { 'IORegistryEntryName': 'IOUSBHostInterface',
                                 'idProduct': 33424,
                                 'idVendor': 1452},
                               { 'IORegistryEntryName': 'IOUSBHostInterface',
                                 'idProduct': 33424,
                                 'idVendor': 1452},
                               { 'IORegistryEntryName': 'IOUSBHostInterface',
                                 'idProduct': 33424,
                                 'idVendor': 1452}],
  'IORegistryEntryName': 'Bluetooth USB Host Controller',
  'idProduct': 33424,
  'idVendor': 1452}
DEBUG:mbedls.lstools_darwin:finding in 
{ 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'AppleUSBHostLegacyClient'},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDUserClient'},
                                                                                                                                                                                          { 'IORegistryEntryName': 'IOHIDParamUserClient'}],
                                                                                                                                                             'IORegistryEntryName': 'IOHIDSystem'}],
                                                                                                                              'IORegistryEntryName': 'AppleDeviceManagementHIDEventService'}],
                                                                                               'IORegistryEntryName': 'IOHIDInterface'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'}],
                                                                'IORegistryEntryName': 'AppleUSBTopCaseHIDDriver',
                                                                'idVendor': 1452}],
                                 'IORegistryEntryName': 'Device Management',
                                 'idProduct': 628,
                                 'idVendor': 1452},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDUserClient'},
                                                                                                                                                                                                                         { 'IORegistryEntryName': 'IOHIDParamUserClient'}],
                                                                                                                                                                                            'IORegistryEntryName': 'IOHIDSystem'}],
                                                                                                                                                             'IORegistryEntryName': 'IOHIDKeyboard'},
                                                                                                                                                           { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDUserClient'},
                                                                                                                                                                                                                         { 'IORegistryEntryName': 'IOHIDParamUserClient'}],
                                                                                                                                                                                            'IORegistryEntryName': 'IOHIDSystem'}],
                                                                                                                                                             'IORegistryEntryName': 'IOHIDConsumer'},
                                                                                                                                                           { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDUserClient'},
                                                                                                                                                                                          { 'IORegistryEntryName': 'IOHIDParamUserClient'}],
                                                                                                                                                             'IORegistryEntryName': 'IOHIDSystem'}],
                                                                                                                              'IORegistryEntryName': 'AppleHIDKeyboardEventDriverV2'}],
                                                                                               'IORegistryEntryName': 'IOHIDInterface'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'}],
                                                                'IORegistryEntryName': 'AppleUSBTopCaseHIDDriver',
                                                                'idVendor': 1452}],
                                 'IORegistryEntryName': 'Keyboard / Boot',
                                 'idProduct': 628,
                                 'idVendor': 1452},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDUserClient'},
                                                                                                                                                                                                                         { 'IORegistryEntryName': 'IOHIDParamUserClient'}],
                                                                                                                                                                                            'IORegistryEntryName': 'IOHIDSystem'}],
                                                                                                                                                             'IORegistryEntryName': 'IOHIDPointing'},
                                                                                                                                                           { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDUserClient'},
                                                                                                                                                                                          { 'IORegistryEntryName': 'IOHIDParamUserClient'}],
                                                                                                                                                             'IORegistryEntryName': 'IOHIDSystem'},
                                                                                                                                                           { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'AppleMultitouchDeviceUserClient'}],
                                                                                                                                                             'IORegistryEntryName': 'AppleMultitouchDevice'}],
                                                                                                                              'IORegistryEntryName': 'AppleMultitouchTrackpadHIDEventDriver'}],
                                                                                               'IORegistryEntryName': 'IOHIDInterface'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'}],
                                                                'IORegistryEntryName': 'AppleUSBTopCaseHIDDriver',
                                                                'idVendor': 1452}],
                                 'IORegistryEntryName': 'Trackpad / Boot',
                                 'idProduct': 628,
                                 'idVendor': 1452},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'AppleActuatorDeviceUserClient'}],
                                                                                                                                                             'IORegistryEntryName': 'AppleActuatorDevice'}],
                                                                                                                              'IORegistryEntryName': 'AppleActuatorHIDEventDriver'}],
                                                                                               'IORegistryEntryName': 'IOHIDInterface'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'}],
                                                                'IORegistryEntryName': 'AppleUSBTopCaseHIDDriver',
                                                                'idVendor': 1452}],
                                 'IORegistryEntryName': 'Actuator',
                                 'idProduct': 628,
                                 'idVendor': 1452},
                               { 'IORegistryEntryChildren': [ { 'IORegistryEntryChildren': [ { 'IORegistryEntryName': 'IOHIDInterface'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'},
                                                                                             { 'IORegistryEntryName': 'IOHIDLibUserClient'}],
                                                                'IORegistryEntryName': 'AppleUSBTopCaseHIDDriver',
                                                                'idVendor': 1452}],
                                 'IORegistryEntryName': 'Accelerometer',
                                 'idProduct': 628,
                                 'idVendor': 1452},
                               { 'IORegistryEntryName': 'AppleUSBHostCompositeDevice'}],
  'IORegistryEntryName': 'Apple Internal Keyboard / Trackpad',
  'USB Serial Number': 'D3H63420U81GHMFA64FF',
  'idProduct': 628,
  'idVendor': 1452}
DEBUG:mbedls.lstools_darwin:_volumes return {'disk2': {'tty': None, 'serial': None, 'vendor_id': 3368, 'product_id': 516}}
DEBUG:mbedls.lstools_darwin:disks dict 
{ 'AllDisks': ['disk0', 'disk0s1', 'disk0s2', 'disk0s3', 'disk1', 'disk2'],
  'AllDisksAndPartitions': [ { 'Content': 'GUID_partition_scheme',
                               'DeviceIdentifier': 'disk0',
                               'Partitions': [ { 'Content': 'EFI',
                                                 'DeviceIdentifier': 'disk0s1',
                                                 'DiskUUID': '402218BA-93DE-4A81-AC3F-9879076C013F',
                                                 'Size': 209715200,
                                                 'VolumeName': 'EFI',
                                                 'VolumeUUID': '0E239BC6-F960-3107-89CF-1C97F78BB46B'},
                                               { 'Content': 'Apple_CoreStorage',
                                                 'DeviceIdentifier': 'disk0s2',
                                                 'DiskUUID': '9E76C9E8-F738-4E0D-A596-CF3CAEEA0084',
                                                 'Size': 499418034176},
                                               { 'Content': 'Apple_Boot',
                                                 'DeviceIdentifier': 'disk0s3',
                                                 'DiskUUID': '9574F941-37C0-4B84-9D0F-D893F473D8CA',
                                                 'Size': 650002432,
                                                 'VolumeName': 'Recovery HD',
                                                 'VolumeUUID': 'C1B4FE5B-234F-3E82-9DE2-87B03AF988ED'}],
                               'Size': 500277790720},
                             { 'Content': 'Apple_HFS',
                               'DeviceIdentifier': 'disk1',
                               'MountPoint': '/',
                               'Size': 499071844352,
                               'VolumeName': 'Macintosh HD'},
                             { 'Content': '',
                               'DeviceIdentifier': 'disk2',
                               'MountPoint': '/Volumes/MBED',
                               'Size': 2121728,
                               'VolumeName': 'MBED'}],
  'VolumesFromDisks': ['Macintosh HD', 'MBED'],
  'WholeDisks': ['disk0', 'disk1', 'disk2']}
DEBUG:mbedls.lstools_base:Candidates for display [{'mount_point': '/Volumes/MBED', 'serial_port': None, 'target_id_usb_id': None}]
WARNING:root:Could not read htm on from usb id None. Falling back to usb id
Traceback (most recent call last):
  File "/usr/local/bin/mbedls", line 11, in <module>
    load_entry_point('mbed-ls==1.3.4', 'console_scripts', 'mbedls')()
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/main.py", line 261, in mbedls_main
    ret_code = args.command(mbeds, args)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/main.py", line 110, in print_table
    return print_mbeds(mbeds, args, False)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/main.py", line 97, in print_mbeds
    devices = mbeds.list_mbeds(unique_names=True, read_details_txt=True)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/lstools_base.py", line 147, in list_mbeds
    }[fs_interaction](device, platform_name_matcher)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/lstools_base.py", line 183, in _fs_before_id_check
    self._update_device_from_htm(device)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/lstools_base.py", line 218, in _update_device_from_htm
    device['platform_name'] = self.plat_db.get(device['target_id'][0:4])
TypeError: 'NoneType' object has no attribute '__getitem__'


```

### PR DESCRIPTION
**Pull request by theotherjimmy**
Resolves #276

### CODE REVIEW

File: test/mbedls_toolsbase.py [hunk 1]
```diff
@@ -69,6 +69,27 @@ def test_list_mbeds_valid_platform(self):
...
+            _mpr.return_value = True
+            _read_htm.side_effect = [(u'0241BEEFDEAD', {}), (None, {})]
+            _get.return_value = 'foo_target'
+            to_check = self.base.list_mbeds()
+            _get.assert_called_once_with('0241')
+        self.assertEqual(len(to_check), 2)
+        self.assertEqual(to_check[0]['target_id'], "0241BEEFDEAD")
+        self.assertEqual(to_check[0]['platform_name'], 'foo_target')
+        self.assertEqual(to_check[1]['target_id'], None)
+        self.assertEqual(to_check[1]['platform_name'], None)
```
Comment (bridadan) [inline test/mbedls_toolsbase.py]: :100: :100: :100: :100: :100: :100: :100: :100: :100: :100: :100: :100: :100:

**Comment by MarceloSalazar** [comment]
Can we proceed to get this merged? We'd like to make sure it's fixed before the 5.7 release. Thanks!

**Comment by MarceloSalazar** [comment]
@theotherjimmy @bridadan are you planning a new release with this change this/next week? Just wanted to make sure this is being shipped in time for the 5.7 release. Thanks!

**Comment by bridadan** [comment]
@theotherjimmy is out of office this week (working just not in the office) but I'll try to get a hold of him

**Comment by MarceloSalazar** [comment]
Bump

**Pull request merged**